math-libs/BLAS/CMakeLists.txt

```
@@ -0,0 +1,296 @@
+    "-DTensile_TEST_LOCAL_PATH=${CMAKE_CURRENT_SOURCE_DIR}/Tensile"
+    # # TODO: Enable clients.
+IGNORE_PACKAGES
+    ROCM
+COMPILER_TOOLCHAIN
+    amd-hip
+BUILD_DEPS
+    therock-msgpack-cxx
+    hipBLAS-common
+RUNTIME_DEPS
+    hip-clr
+    hipBLASLt
+)
+therock_cmake_subproject_glob_c_sources(rocBLAS
+SUBDIRS
+    .
+)
+therock_cmake_subproject_provide_package(rocBLAS rocblas lib/cmake/rocblas)
+therock_cmake_subproject_activate(rocBLAS)
+list(APPEND _blas_subproject_names rocBLAS)
+
+if(THEROCK_ENABLE_SPARSE)
+  ##############################################################################
+  # rocSPARSE
+  ##############################################################################
+  set(_sparse_subproject_names)
+
+  therock_cmake_subproject_declare(rocSPARSE
+    EXTERNAL_SOURCE_DIR "rocSPARSE"
+    BACKGROUND_BUILD
+    CMAKE_ARGS
+      -DHIP_PLATFORM=amd
+      -DROCM_PATH=
+      -DROCM_DIR=
+      -DBUILD_WITH_ROCBLAS=ON
+      -DBUILD_CLIENTS_SAMPLES=OFF
+      -DBUILD_CLIENTS_BENCHMARKS=OFF
+      -DBUILD_CLIENTS_SAMPLES=OFF
+    COMPILER_TOOLCHAIN
+      amd-hip
+    BUILD_DEPS
+      rocBLAS
+      rocPRIM
+    RUNTIME_DEPS
+      hip-clr
+  )
+  therock_cmake_subproject_glob_c_sources(rocSPARSE
+    SUBDIRS
+      .
+  )
+  therock_cmake_subproject_provide_package(rocSPARSE rocsparse lib/cmake/rocsparse)
+  therock_cmake_subproject_activate(rocSPARSE)
+  list(APPEND _sparse_subproject_names rocSPARSE)
+
+  ##############################################################################
+  # hipSPARSE
+  ##############################################################################
+
+  therock_cmake_subproject_declare(hipSPARSE
+    EXTERNAL_SOURCE_DIR "hipSPARSE"
+    BACKGROUND_BUILD
+    CMAKE_ARGS
+      -DHIP_PLATFORM=amd
+      -DROCM_PATH=
+      -DROCM_DIR=
+      -DBUILD_CLIENTS_SAMPLES=OFF
+      -DBUILD_CLIENTS_BENCHMARKS=OFF
+      -DBUILD_CLIENTS_SAMPLES=OFF
+    COMPILER_TOOLCHAIN
+      amd-hip
+    BUILD_DEPS
+      rocSPARSE
+    RUNTIME_DEPS
+      hip-clr
+  )
+  therock_cmake_subproject_glob_c_sources(hipSPARSE
+    SUBDIRS
+      .
+  )
+  therock_cmake_subproject_provide_package(hipSPARSE hipsparse lib/cmake/hipsparse)
+  therock_cmake_subproject_activate(hipSPARSE)
+  list(APPEND _blas_subproject_names hipSPARSE)  
+endif(THEROCK_ENABLE_SPARSE)
+
+
+if(THEROCK_ENABLE_SOLVER)
+  ##############################################################################
+  # rocSOLVER
+  ##############################################################################
+
+  therock_cmake_subproject_declare(rocSOLVER
+      EXTERNAL_SOURCE_DIR "rocSOLVER"
+      BACKGROUND_BUILD
+      CMAKE_ARGS
+      -DHIP_PLATFORM=amd
+      -DROCM_PATH=
+      -DROCM_DIR=
+      # Requires LAPACK
+      -DBUILD_CLIENTS_BENCHMARKS=OFF
+      -DBUILD_CLIENTS_TESTS=OFF
+      COMPILER_TOOLCHAIN
+      amd-hip
+      BUILD_DEPS
+      rocBLAS
+      rocPRIM
+      therock-fmt
+      RUNTIME_DEPS
+      hip-clr
```

CMake formatting (indentation)
```suggestion
      CMAKE_ARGS
        -DHIP_PLATFORM=amd
        -DROCM_PATH=
        -DROCM_DIR=
        # Requires LAPACK
        -DBUILD_CLIENTS_BENCHMARKS=OFF
        -DBUILD_CLIENTS_TESTS=OFF
      COMPILER_TOOLCHAIN
        amd-hip
      BUILD_DEPS
        rocBLAS
        rocPRIM
        therock-fmt
      RUNTIME_DEPS
        hip-clr
```